api4/user.go

```diff
@@ -84,6 +85,67 @@ func getUser(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func getUsers(c *Context, w http.ResponseWriter, r *http.Request) {
```

Comment:
Actually I guess if it's query parameters that could be used for permissions because we can read that easy enough. I just don't like that this function is a giant if statement that doesn't actually support combinations of options. But if this is consistent with others that do, then that's fine. 

Just wanted to bring this up. 2/5 so do whatever you think is best. You have been thinking about this more than me.